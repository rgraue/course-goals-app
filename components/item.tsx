import { Pressable, View, Text } from "react-native";
import { itemStyle } from '@styles';
import { useState } from "react";
import { GoalModal } from "./modal";

export const Item = ({goal, id, goalDeleteHandler}) => {
  const [isModalVisible, switchModalVisible] = useState(false);

  function swapModalView (id) {
    console.log(isModalVisible, id);
    switchModalVisible( currentValue => !currentValue);
  }

  return (
    <View>
      <Pressable 
        onPress={swapModalView.bind(this, goal.item.name)}
        style={ ({pressed}) => pressed && itemStyle.pressedItem}
      >
        <View style={itemStyle.container}>
          <Text>{goal.item.name}</Text>
        </View>
      </Pressable>
      <GoalModal 
        goal={goal.item.name}
        isVisible={isModalVisible}
        closeModalHandler={swapModalView}
        deleteGoalHandler={goalDeleteHandler}
        id={id}
      />
    </View>
  )
}