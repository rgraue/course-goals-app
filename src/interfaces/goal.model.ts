export interface Goal {
  id: string;
  name: string;
}
