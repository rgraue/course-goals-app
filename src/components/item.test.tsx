import React from 'react';
import { ListRenderItemInfo } from 'react-native';
import { render } from '@testing-library/react-native';
import { Item } from './item';
import { Goal } from '../interfaces';

const mockFunc = jest.fn();

const mockGoal: Partial<ListRenderItemInfo<Goal>> = {
  item: {
    name: 'test goal name',
    id: '2',
  },
};

test('GoalItem Component', async () => {
  const tree = render(
    <Item id={mockGoal.item?.id} goal={mockGoal} deleteHandler={mockFunc} />,
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
