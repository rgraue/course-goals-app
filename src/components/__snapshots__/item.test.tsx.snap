// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalItem Component 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "borderRadius": 10,
          "color": "black",
          "fontSize": 18,
          "marginBottom": 10,
          "padding": 10,
          "textAlign": "left",
        }
      }
    >
      <Text>
        test goal name
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</View>
`;
